src/hera/workflows/_mixins.py

```diff
@@ -664,10 +664,12 @@ def __call__(self, *args, **kwargs) -> Union[None, Step, Task]:
                     f"Callable Template '{self.name}' is not callable under a Workflow"  # type: ignore
                 )
             if isinstance(_context.pieces[-1], (Steps, Parallel)):
-                return Step(*args, template=self, **kwargs)
+                allowed_step_kwargs = Step.__fields__.keys()
+                return Step(template=self, **{k: v for k, v in kwargs.items() if k in allowed_step_kwargs})
 
             if isinstance(_context.pieces[-1], DAG):
-                return Task(*args, template=self, **kwargs)
+                allowed_task_kwargs = Task.__fields__.keys()
+                return Task(template=self, **{k: v for k, v in kwargs.items() if k in allowed_task_kwargs})
```

Comment:
Required for otherwise there are extra fields that are not accepeted

Reply:
When would there be extra fields? Do we have them in our examples? Considering

> raise an error when the user specification is incorrect

from the issue, this won't raise an error afaict, and will silently ignore extra kwargs

Reply:
> When would there be extra fields? Do we have them in our examples? Considering

The **kwargs of scripts used to go into the step / task, so that contains extra fields like `image` relative to the fields of step / task. First thing I did after adding the `Extra.forbid` was to run tests. Almost _all_ of the tests failed. 

> from the issue, this won't raise an error afaict, and will silently ignore extra kwargs

Correct but it does with the addition of `Extra.forbid`

Reply:
Ah I see, TIL!